drivers/hdc1000/hdc1000.c

```diff
@@ -132,3 +140,27 @@ int hdc1000_read(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
     xtimer_usleep(HDC1000_CONVERSION_TIME);
     return hdc1000_get_results(dev, temp, hum);
 }
+
+
+int hdc1000_read_cached(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
+{
+    uint32_t now = xtimer_now_usec();
+
+    /* check if initialization phase or outdated */
+    if ((last_read_time == 0xFFFFFFFF && hum_cached == 0 && temp_cached == 0) ||
```

Comment:
I just had one more idea: why not simply do one read at the end of the init function. This way, this check can be omitted and we save these three comparisons...

But this might also be addressed in a follow up - I don't see this as blocker.